### PR TITLE
Better management of unreleased HISTORY

### PR DESCRIPTION
**Pull request by pdillinger**
Summary: See new NOTE in HISTORY.md and unreleased_history/README.txt

Test Plan: some manual testing on my CentOS 8 system

### CODE REVIEW

File: build_tools/release_history.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+function process_file () {
+  # use awk to correct extra or missing newlines, missing '* ' on first line
+  awk '/./ { if (notfirstline || $1 == "*") print;
+             else print "* " $0;
+             notfirstline=1; }' < $1 >> HISTORY.new
+  # Part of the reason for putting the header in a file in the dir is because
+  # git won't track empty directories, but we want the directories to stick
+  # around even when there are no unreleased entries for the group.
+  if echo "$1" | grep -q 000_HEADER.md; then
```
Comment (jaykorean) [inline build_tools/release_history.sh:41]: wondering if we could use `.gitkeep` file for git to track the empty directories.

Comment (pdillinger) [reply build_tools/release_history.sh:41]: Wasn't aware of that. I can do that to simplify some things, but not a big change.

File: unreleased_history/README.txt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+When adding release notes for the next release, add a file to one of these
+directories:
+
+unreleased_history/new_features
+unreleased_history/behavior_changes
+unreleased_history/public_api_changes
+unreleased_history/bug_fixes
+
+with a unique name that makes sense for your or your change, preferably using
```
Comment (jaykorean) [inline unreleased_history/README.txt:12]: typo?

File: unreleased_history/new_features/wait_for_compact.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Add `WaitForCompact()` to wait for all flush and compactions jobs to finish. Jobs to wait include the unscheduled (queued, but not scheduled yet).
```
Comment (jaykorean) [inline unreleased_history/new_features/wait_for_compact.md:1]: I guess `*` is missing on purpose to test the case that missing`*` will be automatically added.

File: unreleased_history/add.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+if [ "$1" ]; then
+  # Target file specified on command line
+  TARGET="$1"
+else
+  # Interactively choose a group and file name
+  DIRS="`find unreleased_history/ -mindepth 1 -maxdepth 1 -type d`"
+  echo "Choose a group for new release note:"
+  echo "$DIRS" | grep -nEo '[^/]+$'
+  echo -n "Enter a number: "
```
[the remark this answers is not in the document]
Comment (pdillinger) [reply unreleased_history/add.sh:14]: Did you try it?

[the remark this answers is not in the document]
Comment (jaykorean) [reply unreleased_history/add.sh:14]: No, not yet. I can try this later today and let you know how it goes.

[the remark this answers is not in the document]
Comment (jaykorean) [reply unreleased_history/add.sh:14]: Ah, it's obvious after trying it myself :)

**Review by jaykorean — APPROVED** [review]
I tried myself and it was easy to use, super convenient. Thank you!

File: unreleased_history/new_features/wait_for_compact.md (hunk 3 above)
Comment (pdillinger) [reply unreleased_history/new_features/wait_for_compact.md:1]: Yes :)

File: unreleased_history/README.txt (hunk 2 above)
Comment (pdillinger) [reply unreleased_history/README.txt]: 👍 Fixed.

**Pull request closed without merging**